### PR TITLE
small UX improvements:

### PR DESCRIPTION
**Pull request by elbayaaa**
#### Which issue(s) does this change fix?
#1713 

#### Why is this change necessary?
Improving `sam pipeline` UX

#### How does it address the issue?
1. show a message when the user cancels a bootstrapping command.
2. Don't prompt for CI/CD provider or provider templates if there is only one choice.
3. Make PipelineFileAlreadyExistsError a UserError.
4. use the Colored class instead of fg='color' when prompting a colored message.
5. Fix a bug where we were not allowing empty response for not required questions.

#### What side effects does this change have?
N/A

#### Checklist

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [x] Write unit tests
- [ ] Write/update functional tests
- [ ] Write/update integration tests
- [x] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

**Review by hoffa — APPROVED** [review]
(Checks failing, otherwise LGTM!)

**Review by aahung — APPROVED** [review]
LGTM

File: samcli/commands/exceptions.py [hunk 1]
```diff
@@ -72,3 +74,10 @@ class AppPipelineTemplateManifestException(UserException):
     Exception class when SAM is not able to parse the "manifest.yaml" file located in the SAM pipeline templates
     Git repo: "github.com/aws/aws-sam-cli-pipeline-init-templates.git
     """
+
+
+class PipelineFileAlreadyExistsError(UserException):
```
Comment (xazhao) [inline samcli/commands/exceptions.py:79]: Might add a brief comment here just like all exceptions above?

File: samcli/lib/pipeline/bootstrap/environment.py [hunk 2]
```diff
@@ -138,7 +140,7 @@ def bootstrap(self, confirm_changeset: bool = True) -> bool:
 
         if self.did_user_provide_all_required_resources():
             click.secho(
-                f"\nAll required resources for the {self.name} environment exist, skipping creation.", fg="yellow"
+                self.color.yellow(f"\nAll required resources for the {self.name} environment exist, skipping creation.")
```
Comment (xazhao) [inline samcli/lib/pipeline/bootstrap/environment.py:143]: Just curious are there any differences between using `self.color.yellow` and `fg="yellow"`?

Comment (elbayaaa) [reply samcli/lib/pipeline/bootstrap/environment.py:143]: No difference in the functionality. Just cleaner code.

**Review by xazhao — APPROVED** [review]
LGTM!

**Pull request merged**